### DISCUSSION TITLE
Relationship properties question?

### DISCUSSION THREAD

**Discussion by vd3d** [flutterdata/flutter_data#194] [Q&A]
Hi,

After searching the documentation, I still have some doubts and wish to ask you some questions please:

- Can I use standard collections like List<> or HasMany/BelongTo are mandatory ?
- Does HasMany is ordered, is it guaranteed to keep the same order ?
- Is there some extensions with methods like: insert, indexOf, etc

Thanks

**Comment by frank06** [comment]
Hi,

 - If you want Flutter Data to automatically handle your relationships you have to use `HasMany`/`BelongsTo`
 - No, `HasMany` does not guarantee order but typically will keep the order by which resources were returned from the API
 - Feel free to call `toList()` on the relationship
